### PR TITLE
feat(dew): the secret resource support new fields

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

The secret resource support new fields.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.The new field contain "secret_type", "enterprise_project_id", "event_subscriptions"
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecret_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_basic -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_basic
=== PAUSE TestAccDewCsmsSecret_basic
=== CONT  TestAccDewCsmsSecret_basic
--- PASS: TestAccDewCsmsSecret_basic (40.20s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       40.254s
```
```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecret_associatedEvent"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_associatedEvent -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_associatedEvent
=== PAUSE TestAccDewCsmsSecret_associatedEvent
=== CONT  TestAccDewCsmsSecret_associatedEvent
--- PASS: TestAccDewCsmsSecret_associatedEvent (120.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       120.859s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 1]
```diff
@@ -121,11 +139,14 @@ func resourceCsmsSecretCreate(ctx context.Context, d *schema.ResourceData, meta
...
-		Description:  d.Get("description").(string),
-		SecretString: d.Get("secret_text").(string),
-		SecretBinary: d.Get("secret_binary").(string),
+		Name:                name,
+		KmsKeyID:            d.Get("kms_key_id").(string),
+		Description:         d.Get("description").(string),
+		SecretString:        d.Get("secret_text").(string),
+		SecretBinary:        d.Get("secret_binary").(string),
+		SecretType:          d.Get("secret_type").(string),
+		EnterpriseProjectID: d.Get("enterprise_project_id").(string),
```
Comment (jinyangyang222) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go:148]: Using the public methods in the config package

File: docs/resources/csms_secret.md [hunk 2]
```diff
@@ -74,6 +95,16 @@ The following arguments are supported:
 
 * `description` - (Optional, String) The description of a secret.
 
+* `secret_type` - (Optional, String, ForceNew) Specifies the type of the secret.
+  Currently, only supported **COMMON**. The default value is **COMMON**。
+
+* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID to which the secret belongs.
+  The default value is **0**.
```
Comment (jinyangyang222) [inline docs/resources/csms_secret.md:102]:
```suggestion
  If omitted, the default enterprise project will be used.
```

File: docs/resources/csms_secret.md [hunk 3]
```diff
@@ -43,6 +43,27 @@ resource "huaweicloud_csms_secret" "test3" {
 }
 ```
 
+### The secret associated event
+
+```hcl
+variable "name" {}
+variable "type" {}
```
Comment (jinyangyang222) [inline docs/resources/csms_secret.md:50]:
```suggestion
variable "secret_type" {}
```

File: docs/resources/csms_secret.md [hunk 4]
```diff
@@ -43,6 +43,27 @@ resource "huaweicloud_csms_secret" "test3" {
...
+variable "secret_string" {}
```
Comment (jinyangyang222) [inline docs/resources/csms_secret.md:51]:
```suggestion
variable "secret_text" {}
```

File: docs/resources/csms_secret.md [hunk 5]
```diff
@@ -43,6 +43,27 @@ resource "huaweicloud_csms_secret" "test3" {
...
+
+resource "huaweicloud_csms_event" "test" {
+  ...
+}
+
+resource "huaweicloud_csms_secret" "test" {
+  name                  = var.name
+  secret_type           = var.type
+  secret_text           = var.secret_string
+  enterprise_project_id = var.eps_id
```
Comment (jinyangyang222) [inline docs/resources/csms_secret.md:62]:
```suggestion
```

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecret_associatedEvent"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_associatedEvent -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_associatedEvent
=== PAUSE TestAccDewCsmsSecret_associatedEvent
=== CONT  TestAccDewCsmsSecret_associatedEvent
--- PASS: TestAccDewCsmsSecret_associatedEvent (112.90s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       112.965s
```

File: docs/resources/csms_secret.md [hunk 6]
```diff
@@ -43,6 +43,25 @@ resource "huaweicloud_csms_secret" "test3" {
...
+variable "secret_text" {}
+
+resource "huaweicloud_csms_event" "test" {
+  ...
+}
+
+resource "huaweicloud_csms_secret" "test" {
+  name                  = var.name
+  secret_type           = var.type
+  secret_text           = var.secret_string
```
Comment (luoping-12345) [inline docs/resources/csms_secret.md]:
```suggestion
  secret_type           = var.secret_type
  secret_text           = var.secret_text
```

File: docs/resources/csms_secret.md [hunk 7]
```diff
@@ -74,6 +93,16 @@ The following arguments are supported:
 
 * `description` - (Optional, String) The description of a secret.
 
+* `secret_type` - (Optional, String, ForceNew) Specifies the type of the secret.
+  Currently, only supported **COMMON**. The default value is **COMMON**。
```
Comment (luoping-12345) [inline docs/resources/csms_secret.md]:
```suggestion
  Currently, only supported **COMMON**. The default value is **COMMON**.
```

File: huaweicloud/utils/utils.go [hunk 8]
```diff
@@ -556,3 +556,32 @@ func IsUUID(uuid string) bool {
...
+	return result
+}
+
+// The method can used to check element is empty or not
+func isNonEmpty(v interface{}) bool {
+	switch v := v.(type) {
+	case string:
+		return v != ""
+	case int, float64, bool:
+		// 对于数值型和布尔类型，非零即为非空
```
Comment (luoping-12345) [inline huaweicloud/utils/utils.go]: Please use English comments.

File: docs/resources/csms_secret.md [hunk 9]
```diff
@@ -43,6 +43,25 @@ resource "huaweicloud_csms_secret" "test3" {
...
+
+resource "huaweicloud_csms_event" "test" {
+  ...
+}
+
+resource "huaweicloud_csms_secret" "test" {
+  name                  = var.name
+  secret_type           = var.type
+  secret_text           = var.secret_string
+  event_subscriptions   = [huaweicloud_csms_event.test.name]
```
Comment (profoundwu) [inline docs/resources/csms_secret.md]:
```suggestion
  event_subscriptions = [huaweicloud_csms_event.test.name]
```
Remove extra spaces

File: huaweicloud/utils/utils.go [hunk 10]
```diff
@@ -556,3 +556,32 @@ func IsUUID(uuid string) bool {
...
+	result := make([]interface{}, 0, len(s))
+	for _, elem := range s {
+		if isNonEmpty(elem) {
+			result = append(result, elem)
+		}
+	}
+	return result
+}
+
+// The method can used to check element is empty or not
```
Comment (profoundwu) [inline huaweicloud/utils/utils.go]:
```suggestion
// The method can be used to check if an element is empty or not.
```
There are syntax issues present, please make adjustments in similar areas

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecret_associatedEvent"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_associatedEvent -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_associatedEvent
=== PAUSE TestAccDewCsmsSecret_associatedEvent
=== CONT  TestAccDewCsmsSecret_associatedEvent
--- PASS: TestAccDewCsmsSecret_associatedEvent (112.23s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       112.295s
```

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 11]
```diff
@@ -89,6 +89,24 @@ func ResourceCsmsSecret() *schema.Resource {
...
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"event_subscriptions": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
```
Comment (jinyangyang222) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go:107]: This field will be backfilled in the read method, please add `Computed` attribute

Comment (ruwenqiang123) [reply huaweicloud/services/dew/resource_huaweicloud_csms_secret.go:108]: the value can be left blank, not need computed

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecret_associatedEvent"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_associatedEvent -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_associatedEvent
=== PAUSE TestAccDewCsmsSecret_associatedEvent
=== CONT  TestAccDewCsmsSecret_associatedEvent
--- PASS: TestAccDewCsmsSecret_associatedEvent (110.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       110.928s
```

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 12]
```diff
@@ -121,11 +140,14 @@ func resourceCsmsSecretCreate(ctx context.Context, d *schema.ResourceData, meta
...
-		SecretString: d.Get("secret_text").(string),
-		SecretBinary: d.Get("secret_binary").(string),
+		Name:                name,
+		KmsKeyID:            d.Get("kms_key_id").(string),
+		Description:         d.Get("description").(string),
+		SecretString:        d.Get("secret_text").(string),
+		SecretBinary:        d.Get("secret_binary").(string),
+		SecretType:          d.Get("secret_type").(string),
+		EnterpriseProjectID: cfg.GetEnterpriseProjectID(d),
+		EventSubscriptions:  utils.ExpandToStringListBySet(d.Get("event_subscriptions").((*schema.Set))),
```
Comment (deer-hang) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: Duplicate parentheses.

**Comment by ruwenqiang123** [comment]
```

root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (dev_eps)$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDewCsmsSecret_associatedEvent"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_associatedEvent -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_associatedEvent
=== PAUSE TestAccDewCsmsSecret_associatedEvent
=== CONT  TestAccDewCsmsSecret_associatedEvent
--- PASS: TestAccDewCsmsSecret_associatedEvent (114.27s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       114.352s
```

**Pull request merged**